### PR TITLE
Allow get repository commits in mock

### PR DESCRIPTION
**Pull request by SamuelTrottier**

### CODE REVIEW

File: NGitLab.Mock/Repository.cs [hunk 1]
```diff
@@ -323,7 +323,7 @@ public IEnumerable<Commit> GetBranchCommits(string branchName)
             }
         }
 
-        public IEnumerable<Commit> GetCommits(string @ref)
+        public IEnumerable<Commit> GetCommits(string @ref = "")
```
Comment (meziantou) [inline NGitLab.Mock/Repository.cs]: I think this is a breaking change (the CI will confirm :)). You need to create a new overload without the parameter. Also, prefer `null` to `""`

File: NGitLab.Mock/Clients/RepositoryClient.cs [hunk 2]
```diff
@@ -87,5 +97,9 @@ public IEnumerable<Ref> GetCommitRefs(Sha1 sha, CommitRefType type = CommitRefTy
         {
             throw new NotImplementedException();
         }
+        private Commit ConvertToNgitlabCommit(LibGit2Sharp.Commit commit, Project project)
```
Comment (meziantou) [inline NGitLab.Mock/Clients/RepositoryClient.cs]:
```suggestion

        private Commit ConvertToNGitLabCommit(LibGit2Sharp.Commit commit, Project project)
```

File: NGitLab.Mock/Repository.cs (hunk 1 above)
Comment (SamuelTrottier) [reply NGitLab.Mock/Repository.cs]: Got it, I was not sure if it was better to add a new overload, i'll add one!

File: NGitLab.Mock/Repository.cs [hunk 3]
```diff
@@ -323,6 +323,11 @@ public IEnumerable<Commit> GetBranchCommits(string branchName)
             }
         }
 
+        public IEnumerable<Commit> GetCommits()
+        {
+            return GetCommits(string.Empty);
```
Comment (meziantou) [inline NGitLab.Mock/Repository.cs:328]:
```suggestion
            return GetCommits(null);
```

Comment (SamuelTrottier) [reply NGitLab.Mock/Repository.cs:328]: I don't think I can make it null, it would be ambiguous which overload I'm trying to call.

**Pull request merged**